lib/signalfx/signalflow/computation.rb

```diff
@@ -0,0 +1,74 @@
+# Represents a SignalFlow computation.  A computation can have a channel
+# associated with it, but not necessarily (it could have been detached while
+# the computation is still running).  New instances should only be created by
```

Comment:
I don't think we actually want those semantics. A `Computation` object is really only meaningful if it is attached to a channel. Which means you should be attaching to a channel in the constructor, like we do here: https://github.com/signalfx/signalfx-python/blob/master/signalfx/signalflow/computation.py#L34

Reply:
It seems more natural and overall simpler to model a Computation as the actual job running in SignalFlow, whether attached to or not, especially in light of the new functionality that allows you to attach to an existing job.  That way, those methods (`attach` and `stop`) can be implemented on this class (see below) and there is a single object that users have to deal with instead of both a computation instance and a handle object.  The `start` command on the client would return an unattached Computation instead of a handle.

Reply:
The handle is not an object. It's literally just a string identifier for the computation. The expected API is, roughly (in Java-isms):

```
Computation execute(String program);
ID start(String program);
Computation attach(ID handle);
void stop(ID handle);
```